src/actions/events.js

```diff
@@ -405,6 +495,34 @@ events.on(FILTER_CLEAR, function() {
   tree.unset(['pages', 'repo', 'filter']);
 });
 
+events.on(BUILD_FILTER, function(event) {
+  const {owner, name, value} = event.data;
+  if (value === '') {
+    tree.unset(['pages', 'repo', owner, name, 'filter']);
+  } else {
+    tree.set(['pages', 'repo', owner, name, 'filter'], value);
+  }
```

Comment:
There should only be a handful of possible children in `['pages']`, one for each page, which means we should have multiple entries for each repository:

```diff
-['pages', 'repo', owner, name, 'filter']
+['pages', 'repo', 'filter']
```

since only a single repository can be viewed at once, I'm not sure it makes sense to track the filter values per repository. Instead, I would suggest it is tracked in one place globally, and reset when the repository changes.